common/test/acceptance/tests/studio/test_studio_grading.py

```diff
@@ -0,0 +1,96 @@
+"""
+Acceptance tests for grade settings in Studio.
+"""
+from common.test.acceptance.pages.studio.settings_graders import GradingPage
+from common.test.acceptance.tests.studio.base_studio_test import StudioCourseTest
+from common.test.acceptance.fixtures.course import XBlockFixtureDesc
+
+
+class GradingPageTest(StudioCourseTest):
+    """
+    Bockchoy tests to add/edit grade settings in studio.
+    """
+
+    url = None
+
+    def setUp(self):  # pylint: disable=arguments-differ
+        super(GradingPageTest, self).setUp()
+        self.grading_page = GradingPage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+
+        self.grading_page.visit()
+
+    def populate_course_fixture(self, course_fixture):
+        """
+        Return a test course fixture.
+        """
+        course_fixture.add_children(
+            XBlockFixtureDesc("chapter", "Test Section").add_children(
+                XBlockFixtureDesc("sequential", "Test Subsection").add_children(
+                )
+            )
+        )
+
+    def test_add_grade_range(self):
+        """
+        Scenario: Users can add grading ranges
+            Given I have opened a new course in Studio
+            And I am viewing the grading settings
+            When I add "1" new grade
+            Then I see I now have "3"
+        """
+        for _ in range(1):
+            length = self.grading_page.total_number_of_grades
+            self.grading_page.click_add_grade()
+            self.assertTrue(self.grading_page.is_grade_added(length))
+        self.grading_page.save()
+        self.grading_page.refresh_and_wait_for_load()
+        total_number_of_grades = self.grading_page.total_number_of_grades
+        self.assertEqual(total_number_of_grades, 3)
+
+    def test_staff_can_add_up_to_five_grades_only(self):
+        """
+        Scenario: Users can only have up to 5 grading ranges
+            Given I have opened a new course in Studio
+            And I am viewing the grading settings
+            When I try to add more than 5 grades
+            Then I see I have only "5" grades
+        """
+        for grade_ordinal in range(1, 6):
+            length = self.grading_page.total_number_of_grades
+            self.grading_page.click_add_grade()
+            # By default page has 2 grades, so greater than 3 means, attempt is made to add 6th grade
+            if grade_ordinal > 3:
+                self.assertFalse(self.grading_page.is_grade_added(length))
```

Comment:
This requires waiting for a 5-second timeout (twice).  Is there no better way to detect that the button was successfully clicked and didn't add a new grade?  Just trying to avoid adding 10 seconds to the test duration if we can avoid it.

Reply:
The code was making two attempts to add more than 5 grades, only one attempt would have worked well, but I just tried to make sure that 6th grade does not get added. But now I have reduced it to one attempt only and reduced the timeout as well. So now it will wait for 3 seconds only.